test/integration/channels.js

```diff
@@ -2337,3 +2337,265 @@ describe('partial update channel', () => {
 		}
 	});
 });
+
+describe('Threadless replies', () => {
+	let client, channel, ruud, friend, firstMessage, secondMessage, threadlessReply;
+
+	before(async () => {
+		ruud = 'ruud-' + uuidv4();
+		friend = 'friend-' + uuidv4();
+		await createUsers([ruud, friend]);
+		client = await getTestClient(true);
+		channel = client.channel('messaging', uuidv4(), {
+			created_by_id: ruud,
+			members: [ruud, friend],
+		});
+		await channel.create();
+	});
+
+	after(async () => {
+		await channel.delete();
+		await client.deleteUser(ruud, { hard_delete: true });
+		await client.deleteUser(friend, { hard_delete: true });
+	});
+
+	describe('Ruud sends a message in the channel', () => {
+		it('is possible to create regular messages', async () => {
+			const res = await channel.sendMessage({
+				text: 'The one message to rule them all',
+				user_id: ruud,
+			});
+			const res2 = await channel.sendMessage({
+				text: 'The second message to rule all, except the first message',
+				user_id: ruud,
+			});
+			firstMessage = res.message;
+			secondMessage = res2.message;
+		});
+	});
+
+	describe('Friend replies to the message in a thread', () => {
+		it('is possible to reply to the message', async () => {
+			const res = await channel.sendMessage({
+				text: 'The first threaded reply',
+				user_id: friend,
+				parent_id: firstMessage.id,
+			});
+
+			expect(res.message).to.not.be.undefined;
+			expect(res.message.id).to.not.be.undefined;
+			expect(res.message.id).to.not.be.empty;
+			expect(res.message.parent_id).to.not.be.undefined;
+			expect(res.message.type).to.equal('reply');
+		});
+	});
+
+	describe('Friend sends a threadless / quoted reply', () => {
+		it("is possible to send a reply that isn't in a thread", async () => {
+			const res = await channel.sendMessage({
+				text: 'The first threadless reply',
+				user_id: friend,
+				reply_to_message_id: firstMessage.id,
+			});
+
+			expect(res.message).to.not.be.undefined;
+			expect(res.message.id).to.not.be.undefined;
+			expect(res.message.id).to.not.be.empty;
+			expect(res.message.reply_to_message_id).to.not.be.undefined;
+			expect(res.message.reply_to_message).to.not.be.undefined;
+			expect(res.message.reply_to_message_id).to.equal(firstMessage.id);
+			expect(res.message.reply_to_message.id).to.equal(firstMessage.id);
+			expect(res.message.reply_to_message.text).to.not.be.undefined;
+			expect(res.message.reply_to_message.text).to.equal(firstMessage.text);
+			expect(res.message.type).to.equal('regular');
```

Comment:
if we have an enum for this, would be better to use it

Reply:
I don't think we have this